lib/introspective_grape/utils/json_expander.rb

```diff
@@ -0,0 +1,21 @@
+module IntrospectiveGrape
+  module Utils
+    module JsonExpander
+      extend self
+
+      # Expands an object with its #to_json method if it's not a primitive or if the first element is a Grape::Entity.
+      def expand(object)
+        if (object.respond_to?(:to_json) &&
+          [String, Symbol, Hash, Array, NilClass, TrueClass, FalseClass, Numeric].all? { |x| !object.is_a?(x) }) ||
+          # Force arrays of Grape::Entities into their hash representations before camelizing
+          (object.is_a?(Array) && defined?(Grape::Entity) && object.first.is_a?(Grape::Entity))
+
```

Comment:
> I remember the point of the increasingly complicated logic here was to not do JSON.parse(object.to_json) if the object was already a hash, or an array of hashes. Maybe something like

So if that's the case, shouldn't we just directly check for that so

```ruby
(o.is_a?(Array) && o.first.is_a?(Hash)) || o.is_a?(Hash)
```

I'm not sure what the logic was, so I made a best guess, but based on what you are telling me, we should write the above.

What if the object was a primitive, like a [String, Symbol, Hash, Array, NilClass, TrueClass, FalseClass, Numeric], then we *should* do `JSON.parse(object.to_json)`? If so, why?